src/Pagination/DoctrinePaginatorAdapter.php

```diff
@@ -73,7 +73,7 @@ public function getTotal(): int
      */
     public function getCount(): int
     {
-        return $this->paginator->getIterator()->count();
+        return $this->paginator->count();
```

Comment:
`->getTotal()` returns the total number of results while `->getCount()` should return the number of results in the current page.
```suggestion
        return $this->paginator->getIterator()->count();
```

Reply:
This is a fix from @benepfist, which I wrongfully assumed to be correct.

I reverted the change. Because the documented return type of the base method in Doctrine being `Traversable`, phpstan will complain here. As we know, that this is an `ArrayIterator`, we can skip the warning from phpstan imo. If this changes at some point, it SHOULD throw an error and a new solution has to be found for the to-be implementation.

Reply:
A better solution would be something like:
```php
$iterator = $this->paginator->getIterator();
if (is_countable($iterator)) {
    return count($iterator);
}

return $this->paginator->count();
```

Reply:
today I learned of the [iterator_count()](https://www.php.net/manual/en/function.iterator-count.php) function. Added a new approach using this.

Your's would work aswell; if you prefer this after my suggestion, I will update to your approach.

Reply:
Nice, though I wonder about the implication of seeking the iterator: https://3v4l.org/gQEB5

Reply:
It's a valid point and I thought about always cloning the instance for this reason: 
```php
return iterator_count(clone $this->paginator->getIterator());
```
but I feel like this could be way to expensive for to little benefit.

So I went back to your suggestion checking for `is_countable()` and using `count()`, which with the current Doctrine version uses the `->count()` method on the ArrayIterator:
- If implementation would change to either another `\Countable` iterator or even a plain `array`, this would still be fully functional and performant.
- In case the iterator be replaced with another `\Traversable` that is not implementing `\Countable` this would fallback to use `iterator_count()` with a cloned instance of the iterator - a little more expensive, but the benefits of having the correct result with unaffected iterator (thinking about [NoRewindIterator](https://www.php.net/manual/en/class.norewinditerator.php)) are worth the cost IMO. This path is to handle a rather unlikely fallback case.

I hope you can agree on this suggestion. If not - please provide the implementation you'd like and I will gladly do it that way. None of my projects uses Doctrine anyway, so I'm rather uninformed about the internals.

Reply:
Can you please remove this change and include it in a separate pull request? Thanks.